plugins/outputs/sensu/sensu.go

```diff
@@ -0,0 +1,468 @@
+# agent_api_url = "http://127.0.0.1:3031"
+
+## API KEY is the Sensu Backend API token 
+## Generate a new API token via: 
+## 
+## $ sensuctl cluster-role create telegraf --verb create --resource events,entities
+## $ sensuctl cluster-role-binding create telegraf --cluster-role telegraf --group telegraf
+## $ sensuctl user create telegraf --group telegraf --password REDACTED 
+## $ sensuctl api-key grant telegraf
+##
+## For more information on Sensu RBAC profiles & API tokens, please visit: 
+## - https://docs.sensu.io/sensu-go/latest/reference/rbac/
+## - https://docs.sensu.io/sensu-go/latest/reference/apikeys/ 
+## 
+# api_key = "${SENSU_API_KEY}"
+
+## Optional TLS Config
+# tls_ca = "/etc/telegraf/ca.pem"
+# tls_cert = "/etc/telegraf/cert.pem"
+# tls_key = "/etc/telegraf/key.pem"
+## Use TLS but skip chain & host verification
+# insecure_skip_verify = false
+
+## Timeout for HTTP message
+# timeout = "5s"
+
+## HTTP Content-Encoding for write request body, can be set to "gzip" to
+## compress body or "identity" to apply no encoding.
+# content_encoding = "identity"
+
+## Sensu Event details 
+## 
+## NOTE: if the output plugin is configured to send events to a 
+## backend_api_url and entity_name is not set, the value returned by 
+## os.Hostname() will be used; if the output plugin is configured to send
+## events to an agent_api_url, entity_name and entity_namespace are not used. 
+# [outputs.sensu-go.entity]
+#   name = "server-01"
+#   namespace = "default"
+
+# [outputs.sensu-go.tags]
+#   source = "telegraf"
+
+# [outputs.sensu-go.metrics]
+#   handlers = ["elasticsearch","timescaledb"]
+`
+
+// Description provides a description of the plugin
+func (s *Sensu) Description() string {
+	return "Send aggregate metrics to Sensu Monitor"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (s *Sensu) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Sensu) SetSerializer(serializer serializers.Serializer) {
+	s.serializer = serializer
+}
+
+func (s *Sensu) createClient(ctx context.Context) (*http.Client, error) {
+	tlsCfg, err := s.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: s.Timeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (s *Sensu) Connect() error {
+	if s.Timeout.Duration == 0 {
+		s.Timeout.Duration = defaultClientTimeout
+	}
+
+	ctx := context.Background()
+	client, err := s.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	s.client = client
+
+	return nil
+}
+
+func (s *Sensu) Close() error {
+	return nil
+}
+
+func (s *Sensu) Write(metrics []telegraf.Metric) error {
+	var points []*OutputMetric
+	for _, metric := range metrics {
+		// Add tags from config to each metric point
+		var tagList []*OutputTag
+		for name, value := range s.Tags {
+			metric.AddTag(name, value)
+		}
```

Comment:
I don't like modifying the input metrics. Please add those tags to `tagList` directly! Maybe it would also be nice to pre-allocate the `tagList` as you can expect `len(s.Tags) + len(metric.TagList())` entries.

Reply:
This should be fixed now.